### PR TITLE
Fix LightningCLI loading of hyperparameters from ckpt_path failing for subclass model mode

### PR DESCRIPTION
**Pull request by mauvilsa**
## What does this PR do?

Fixes https://github.com/torchgeo/torchgeo/issues/2995
Maybe also fixes https://github.com/mehta-lab/VisCy/issues/291

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21246.org.readthedocs.build/en/21246/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: src/lightning/pytorch/cli.py [hunk 1]
```diff
@@ -564,6 +564,11 @@ def _parse_ckpt_path(self) -> None:
             hparams.pop("_instantiator", None)
             if not hparams:
                 return
+            if "_class_path" in hparams:
+                hparams = {
+                    "class_path": hparams["_class_path"],
+                    "dict_kwargs": {k: v for k, v in hparams.items() if k != "_class_path"},
+                }
```
Comment (SkafteNicki) [inline src/lightning/pytorch/cli.py:571]:
```suggestion
            if "_class_path" in hparams:
                hparams = {
                    "class_path": hparams.pop("_class_path"),
                    "dict_kwargs": hparams,
                }
```

**Comment by adamjstewart** [comment]
Looks like this didn't make it into 2.5.6. Will there be a 2.5.7 or will 2.6.0 be next?

**Comment by deependujha** [comment]
Hi @adamjstewart, We will probably do 2.6.0 early next week.
